code-challenges.test.js

```diff
@@ -0,0 +1,185 @@
+      expect(capitalization(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+      expect(capitalization(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+
+    })
+  })
+
+// ----->
+//  FAIL  ./code-challenges.test.js
+//   divBy3
+//   ✓ test with expect statements for each of the variables provided. (3 ms)
+// capitalization
+//   ✕ test with expect statements for each of the variables provided.
+
+
+// b) Create the function that makes the test pass.
+// pseudo code: 
+// input: A code that would take an array with strings
+// output: returns every string capitalize only at the first character
+// create a code named capitalization that takes an array as a parameter
+// return the following:
+    //  use the high order function .map() on the array
+    //  use buil-in method .charAt() identifing the first index of each string within the array 
+        // use built-in method .toUpperCase() to capitalize the index requested
+    // then concatenate using '+' (since we are working with the index of the charaters in the string) 
+    // lastly use the built-in method .slice() to select only the first character in the string. 
+
+
+const capitalization = (array) =>{
+    return array.map((value) => value.charAt(0).toUpperCase() + value.slice(1))
+}
+
+// PASS  ./code-challenges.test.js
+// divBy3
+//   ✓ test with expect statements for each of the variables provided. (3 ms)
+// capitalization
+//   ✓ test with expect statements for each of the variables provided. (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+
+
+// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const vowelTester1 = "learn"
+// Expected output: 1
+const vowelTester2 = "academy"
+// Expected output: 0
+const vowelTester3 = "challenges"
+// Expected output: 2
+
+describe("indexFirstVowel", () => {
+        it("takes in a string and logs the index of the first vowel.", () => {
+          expect(indexFirstVowel(vowelTester1)).toEqual(1)
+          expect(indexFirstVowel(vowelTester2)).toEqual(0)
+          expect(indexFirstVowel(vowelTester3)).toEqual(2)
+        })
+      })
+
+// FAIL  ./code-challenges.test.js
+// divBy3
+// ✓ takes a object as an argument and decides if the number inside it is evenly divisible by three or not. (3 ms)
+// capitalization
+// ✓ takes in an array of words and returns an array with all the words capitalized. (1 ms)
+// indexFirstVowel
+// ✕ takes in a string and logs the index of the first vowel.
+    
+
+// b) Create the function that makes the test pass.
+
+
+
+const indexFirstVowel = (string) =>{
+    let vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
+    return string.filter((value, index) =>{
+        return values.indexOf(vowels)})
+    }
+    
```

Comment:
Nice! I like that you left the other approach and built on it by thinking it through and going another way instead. Great job! This could be left so you understand that there was one working approach with a for loop OR a filter.